### DISCUSSION TITLE
Add Mobile to initial MFA with OTP & SMS

### DISCUSSION THREAD

**Discussion by ghost** [community/community#57479] [Mobile]
Github Mobile as 2FA is more secure than OTP (phishing possible) or SMS highly insecure please add it soon

**Comment by HenSquared** [comment]
Hi @BirdInFire , you are able to use the mobile app for 2fa. Here is a guide: https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication#configuring-two-factor-authentication-using-github-mobile

Let me know if this doesn't quite answer your concern. Thanks!

**Reply by ghost** [reply]
> Hi @BirdInFire , you are able to use the mobile app for 2fa. Here is a guide: https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication#configuring-two-factor-authentication-using-github-mobile
> 
> Let me know if this doesn't quite answer your concern. Thanks!

No because the otp itself is vulnerable, exemple, some phishing tool replicate perfectly github login and if you put the otp in it your account is hacked, only an app notification or a security key can prevent these kind of attack (why i do ask to let us chose the app as principal 2FA and so do not accept OTP on our account).

**Reply by HenSquared** [reply]
> GitHub Mobile does not rely on TOTP, and instead uses public-key cryptography to secure your account.

When users have GitHub Mobile configured to use 2FA, after a sign in request, a user will receive a notification prompting them to allow or reject the sign in. 

https://github.blog/wp-content/uploads/2022/01/GitHub-Mobile-2FA-demo.gif

I've attached a link to the demo and here is the blog post about the feature.

https://github.blog/2022-01-25-secure-your-github-account-github-mobile-2fa/

I believe this request is handled by this existing feature.

**Reply by ghost** [reply]
> > GitHub Mobile does not rely on TOTP, and instead uses public-key cryptography to secure your account.
> 
> When users have GitHub Mobile configured to use 2FA, after a sign in request, a user will receive a notification prompting them to allow or reject the sign in.
> 
> https://github.blog/wp-content/uploads/2022/01/GitHub-Mobile-2FA-demo.gif
> 
> I've attached a link to the demo and here is the blog post about the feature.
> 
> https://github.blog/2022-01-25-secure-your-github-account-github-mobile-2fa/
> 
> I believe this request is handled by this existing feature.

Why Github seem hostile to lettings us having more secure 2FA as default (during the enabling (like Gitlab / Apple / Google etc) by continuing to force us to allow unsecure OTP or SMS on our account) ?

Github do it's best to secure against attack on code (and inside code (token scan etc)), but you (as a company) don't seem to understand that you must stop to gate keep the use of secure 2FA (FIDO2 / the tap on github mobile) behind a mandatory OTP / SMS less secure than fido 2 / github mobile.

I mean why actually prevent us to add more security on our account by stopping to gate keep the MFA, and don't tell me it's to prevent to lose our account the emergency code are here for this already

**Comment by DNin01** [comment]
The only problem with that is that if you sign out of GitHub on all of your mobile devices, you'd have no way back in. This is because the GitHub Mobile option is not driven by your device - GitHub just sends a notification to all mobile devices logged into the same GitHub account, unlike TOTP apps and security keys where it's the device that stores the key to logging in.

But GitHub said in a [blog post](https://github.blog/changelog/2022-11-21-updates-to-the-two-factor-authentication-setup-flow/) that they were going to make it possible to register a security key during the initial 2FA setup. That might mean that when the time comes, you'll be able to use only a security key.

**Reply by ghost** [reply]
and the Recovery key for the dog ?

**Comment by yangsfang** [comment]
Both Github Mobile and security keys are considered _secondary_ 2FA methods at the moment. This means you are required to set up a _primary_ 2FA method, which is either TOTP or SMS, before you can use either Github Mobile or security keys. 

Github did plan to make security keys a primary method sometime in 2023, but the priority is not high enough for it to be done before 2FA enforcement is rolled out. So you are forced to use the less secure TOTP or SMS, in combination of Github Mobile or security keys.

What you can do is to set up TOTP as primary 2FA (using third party mobile app, web app, or desktop app), add security keys and Github Mobile as secondary 2FA, then destroy the seeds of the TOTP. It will act like it has ever existed, unless someone managed to phish out the seed before you destroy it.

**Reply by ghost** [reply]
Yeah but it's a boring tricks, i prefer maintain my prposal, remove sms, and replace it with key / mobile, i've seen today the removal of sms from sudo mode, but it must be removed totally, sms is even less secure than email.
